### ISSUE TITLE
Bios A23

### ISSUE THREAD

**Issue by ams0** [bacher09/pwgen-for-bios#166]
I am trying for months now to unlock a Dell E7250 with System Number D803M72-6FF1. The problem is that I have Bios version A23, which I can't find any reference online about. No password work, even replacing 6FF1 with BF97, and I can only get in the graphical BIOS interface, never on the black background one. Anyone is in the same boat as me?

**Comment by K3YL0** [comment]
What passwords have you tried? What type of keyboard are you using?

**Comment by ams0** [comment]
US keyboard. `pNMIk1ky9Qr9MyrZ` and `04GbCsFCV56Bf2Ds` as reported from `https://bios-pw.org/`

**Comment by K3YL0** [comment]
Make sure to pay attention to Zero "0" And Oh's "O" etc. And after entering password, Press CTRL+ENTER or CTRL+ENTER+ENTER. Are you able to set the BIOS to defaults? Maybe need to downgrade BIOS version?

**Comment by ams0** [comment]
Yes, double checked the strings, and I can't downgrade, I need the admin password for that...Do I still have hopes you think?

**Comment by K3YL0** [comment]
Yes, I do believe their is still hope! I will be taking a closer look at your issue later today. Hopefully someone else may help out as well. I have A LOT going on right now, Dealing with COVID and everything else life has thrown at me.

**Comment by ams0** [comment]
Thank you for your time and I sincerely hope things get better for you and everyone.

**Comment by K3YL0** [comment]
Any luck on your end yet? I wish I was their to physically run a USB boot disk on it to test some scripts! ฅ^•ﻌ•^ฅ

**Comment by ams0** [comment]
I’m happy to run those scripts for you and send you results. I’m an engineer and hacking since the tender age of 8 :)

**Comment by K3YL0** [comment]
I'm not ready to release them to the "Wild" 🙃

**Comment by bacher09** [comment]
Actually A23 is available on [manufacturer website](https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=61vx8&oscode=wb64a&productcode=latitude-e7250-ultrabook) and from my quick glimpse it looks the same as code that available on the website, a bit latter I'll check more precisely.

**Comment by bacher09** [comment]
Your password has to be `04GbCsFCV56Bf2Ds` (that's what actually used in `A23`). You can actually check it yourself, with modern tooling you can just find block that executes password execute it with [unicorn](https://github.com/unicorn-engine/unicorn), also you can try [qiling](https://github.com/qilingframework/qiling) (which is wrapper over unicorn and makes few things easier).

**Comment by bacher09** [comment]
CMOS battery never helps on laptops, because password is usually saved in NVRAM. There might be some issues with keyboard (sometimes certain keys don't work, or register as wrong scan code in some cases certain key can be stuck, etc), try attaching external keyboard (and maybe even detaching onboard keyboard).
Also, if you have SPI programmer, you could patch firmware to accept any password or even reset NVRAM.

**Comment by ams0** [comment]
External keyboard tried, and failed. I might find an SPI programmer, any pointer to docs how to do it, would be extremely appreciated. Thanks!

**Comment by K3YL0** [comment]
> External keyboard tried, and failed. I might find an SPI programmer, any pointer to docs how to do it, would be extremely appreciated. Thanks!

This might get you going in the right direction! ฅ^•ﻌ•^ฅ
[https://jensd.be/980/linux/bios-or-spi-programming-on-windows-or-linux-using-a-ch341a](url)
